src/Messages/OWSBlockingManager.m

```diff
@@ -201,6 +214,15 @@ - (void)lazyLoadBlockedPhoneNumbersIfNecessary
                          inCollection:kOWSBlockingManager_BlockedPhoneNumbersCollection];
     _blockedPhoneNumberSet = [[NSMutableSet alloc] initWithArray:(blockedPhoneNumbers ?: [NSArray new])];
 
+    [self syncBlockedPhoneNumbersIfNecessary];
+    [self observeNotifications];
+}
+
+// This method should only be called from within a synchronized block.
```

Comment:
> Do we need to synchronize(self) the call from applicationDidBecomeActive?

Great catch.

> can we make blockedPhoneNumberSet an atomic property since we're potentially reading from it on different threads, and sometimes re-assigning it.

I've made this atomic, but this shouldn't be necessary - it's always accessed within a synchronized block.